chef/cookbooks/utils/providers/line.rb

```diff
@@ -87,14 +87,40 @@ def add_and_filter(file, add, filter_re)
   unlock(lock) unless lock.nil?
 end
 
+def search_and_replace(filename, search_regex, replaced_text)
+  rc = Chef::Util::FileEdit.new(filename)
+  rc.search_file_replace_line(
+    search_regex,
+    replaced_text
+  )
+  res = rc.file_edited?
+  rc.write_file
+  res
+end
+
 action :add do
-  updated = add_and_filter(@new_resource.file,
-			                        @new_resource.name, @new_resource.regexp_exclude)
+  updated = add_and_filter(
+    @new_resource.file,
+    @new_resource.name,
+    @new_resource.regexp_exclude
+  )
   @new_resource.updated_by_last_action(true) if updated
 end
 
 action :remove do
-  updated = add_and_filter(@new_resource.file,nil,
-			                        "^#{@new_resource.name}$")
+  updated = add_and_filter(
+    @new_resource.file,
+    nil,
+    "^#{@new_resource.name}$"
+  )
+  @new_resource.updated_by_last_action(true) if updated
+end
+
+action :replace do
+  updated = search_and_replace(
+    @new_resource.file,
+    @new_resource.regexp,
```

Comment:
@vuntz correct, i don't know what happened, but it's fixed now. Also `:replace` was not defined